### PR TITLE
Add metadata

### LINKED ISSUE #313: Easily adding metadata
As an index-maker, I want to be able to easily add metadata to odinson docs so that it can be easily indexed.

Suggested method would be a parallel json file that can be merged with OdinsonDocument in a separate step in the IndexDocuments.

Indexing on the command line, stick with programs in extra, add new app `UpdateMetadata.scala` -- in place? copy? replace current metadata with new.

### PR DESCRIPTION
**Pull request by BeckySharp**
App for being able to add metadata to an OdinsonDocument

moved the parallel file / relative path computation to `ExtraFileUtils` to be able to use in multiple places.

closes #313

### CODE REVIEW

File: core/src/main/scala/ai/lum/odinson/OdinsonDocument.scala [hunk 1]
```diff
@@ -178,3 +187,17 @@ object NestedField {
   }
 
 }
+
+/** Helper class for reading and writing metadata companion files */
+case class MetadataWrapper(id: String, fields: Seq[Field]) {
```
Comment (BeckySharp) [inline core/src/main/scala/ai/lum/odinson/OdinsonDocument.scala:192]: rename to docId

Comment (BeckySharp) [reply core/src/main/scala/ai/lum/odinson/OdinsonDocument.scala:192]: possibly expand the comment to explain how it's helpful/used

File: extra/src/test/scala/ai/lum/odinson/extra/TestAddMetadataToDocuments.scala [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    AddMetadataToDocuments.usingConfigSettings(settings)
+
+    // verify that the original documents still don't have metadata
+    for (f <- new File(s"${tmpFolder}/docs").listFiles()) {
+      val doc = Document.fromJson(f)
+      doc.metadata.toArray shouldBe empty
+    }
+
+    // verify that the new documents now have metadata and that they're in the right place
+    for (f <- new File(s"${tmpFolder}/docsWithMetadata").listFiles()) {
```
Comment (BeckySharp) [inline extra/src/test/scala/ai/lum/odinson/extra/TestAddMetadataToDocuments.scala:55]: check on the length of the files to ensure not 0

**Review by marcovzla — APPROVED** [review]
Thanks!

**Pull request merged**